### PR TITLE
Add docs for CSV and CatalogSource

### PR DESCRIPTION
**Pull request by ecordell**

### CODE REVIEW

File: content/en/docs/Concepts/crds/CatalogSource.md [hunk 1]
```diff
@@ -3,8 +3,41 @@ title: "CatalogSource"
 weight: 3
 ---
 
-A CatalogSource represents a repository of bundles, each of which is a collection of [CSVs](/docs/concepts/crds/clusterserviceversion), [CRDs](/docs/concepts/crds), and package definition, that represents an operator.
+A CatalogSource represents a store of metadata that OLM can query to discover and install operators and their dependencies.
 
-{{% alert title="Warning" color="warning" %}} 
-This section is under construction.
-{{% /alert %}}
+The `.spec` of a CatalogSource indicates to how to construct a pod or how to talk to a service that serves the [operator-registry grpc api](/docs/operator-registry).
```
Comment (harishsurf) [inline content/en/docs/Concepts/crds/CatalogSource.md:8]: indicates to how to -> indicates how to

File: content/en/docs/Concepts/crds/CatalogSource.md [hunk 2]
```diff
@@ -3,8 +3,41 @@ title: "CatalogSource"
...
-{{% alert title="Warning" color="warning" %}} 
-This section is under construction.
-{{% /alert %}}
+The `.spec` of a CatalogSource indicates to how to construct a pod or how to talk to a service that serves the [operator-registry grpc api](/docs/operator-registry). 
+
+There are three primary types:
+
+ - `grpc` with and `image` referece: OLM will pull the image and run the pod, which is expected to serve a compliant api.
+ - `grpc` with an `address` field: OLM will attempt to contact the grpc api at the given address. This should not be used in most cases.
+ - `internal` or `configmap`: OLM will parse the configmap's data and spin up a pod that can serve the grpc api over it.
```
Comment (harishsurf) [inline content/en/docs/Concepts/crds/CatalogSource.md:14]: I believe `internal` is deprecated. We could skip this or add a note?

File: content/en/docs/Concepts/crds/CatalogSource.md [hunk 3]
```diff
@@ -3,8 +3,41 @@ title: "CatalogSource"
...
+  name: operatorhubio-catalog
+  namespace: olm
+spec:
+  sourceType: grpc
+  image: quay.io/operator-framework/upstream-community-operators:latest
+  displayName: Community Operators
+  publisher: OperatorHub.io
+```
+
+This defines a CatalogSource for OperatorHub.io's content. The `name` of the CatalogSource is used as input to a `Subscription`, which tells OLM where to look to find a requested operator:
```
Comment (harishsurf) [inline content/en/docs/Concepts/crds/CatalogSource.md:31]: We could add reference link for Subscription (/docs/concepts/crds/subscription/)

File: content/en/docs/Concepts/crds/clusterserviceversion.md [hunk 4]
```diff
@@ -3,9 +3,105 @@ title: "ClusterServiceVersion"
 weight: 2
 ---
 
-A ClusterServiceVersion(CSV) represents a particular version of a ClusterService and its operator. It includes metadata such as name, description, version, repository link, labels, icon, etc. It declares `owned`/`required` CRDs, cluster requirements, and install strategy that tells OLM how to create required resources and set up the operator as a [deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/).
+A ClusterServiceVersion (CSV) represents a particular version a running operator on a cluster. It includes metadata such as name, description, version, repository link, labels, icon, etc. It declares `owned`/`required` CRDs, cluster requirements, and install strategy that tells OLM how to create required resources and set up the operator as a [deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/).
```
Comment (harishsurf) [inline content/en/docs/Concepts/crds/clusterserviceversion.md:6]: nit: version **of** a running operator

**Comment by njhale** [comment]
@harishsurf I think these are all solid suggestions, but in the interest of time should we merge and follow up with a fix? The added content will be useful, even if there are a few grammatical mistakes for a brief period.

**Pull request merged**